subsys/usb/device_next/class/usbd_msc_scsi.c

```diff
@@ -536,13 +563,17 @@ SCSI_CMD_HANDLER(MODE_SENSE_6)
 
 	ctx->cmd_is_data_read = true;
 
+	if (cmd->page != MODE_SENSE_PAGE_CODE_ALL_PAGES || cmd->subpage != 0) {
+		return illegal_request(ctx, INVALID_FIELD_IN_CDB);
+	}
+
 	r.mode_data_length = 3;
 	r.medium_type = 0x00;
 	r.device_specific_parameter = 0x00;
 	r.block_descriptor_length = 0x00;
 
 	BUILD_ASSERT(sizeof(r) <= CONFIG_USBD_MSC_SCSI_BUFFER_SIZE);
-	length = MIN(sys_be16_to_cpu(cmd->allocation_length), sizeof(r));
```

Comment:
This change is suspicious, was sys_be16_to_cpu() not necessary?

Reply:
Because MODE SENSE(6) has only 8-bit long allocation length, not 16 bit.